python/ml4ir/base/model/optimizer.py

```diff
@@ -37,38 +24,83 @@ def get_optimizer(
     References:
         https://www.tensorflow.org/api_docs/python/tf/keras/optimizers/Optimizer
         https://www.tensorflow.org/api_docs/python/tf/keras/optimizers/schedules/ExponentialDecay
+        https://arxiv.org/pdf/1506.01186.pdf
 
     FIXME:
         Define all arguments overriding tensorflow defaults in a separate file
         for visibility with ml4ir users
     """
-    # Define an exponential learning rate decay schedule
-    learning_rate_schedule = ExponentialDecay(
-        learning_rate,
-        decay_steps=learning_rate_decay_steps,
-        decay_rate=learning_rate_decay,
-        staircase=True,
-    )
+    model_config = file_io.read_yaml(model_config_file)
 
-    if optimizer_key == OptimizerKey.ADAM:
-        return tf_optimizers.Adam(
-            learning_rate=learning_rate_schedule, clipvalue=gradient_clip_value
-        )
-    elif optimizer_key == OptimizerKey.NADAM:
-        return tf_optimizers.Nadam(
-            learning_rate=learning_rate_schedule, clipvalue=gradient_clip_value
-        )
-    elif optimizer_key == OptimizerKey.ADAGRAD:
-        return tf_optimizers.Adagrad(
-            learning_rate=learning_rate_schedule, clipvalue=gradient_clip_value
-        )
-    elif optimizer_key == OptimizerKey.SGD:
-        return tf_optimizers.SGD(
-            learning_rate=learning_rate_schedule, clipvalue=gradient_clip_value
-        )
-    elif optimizer_key == OptimizerKey.RMS_PROP:
-        return tf_optimizers.RMSprop(
-            learning_rate=learning_rate_schedule, clipvalue=gradient_clip_value
+    if 'lr_schedule' not in model_config:
```

Comment:
I am not onboard with the choice of having `ExponentialDecay` as default scheduler. I prefer a constant learning rate to be the default, i think this is what the community expects too!

Reply:
+1

I have recommended @mohazahran to set the default to "default" / "uniform" / "constant" learning rate schedule.

Reply:
yes. I think we left comments at the same time. I will do a pass over it tomorrow again! To be clear: I think this is a great and much needed PR @mohazahran.